scripts/dir_hash.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python3
+
+# Copyright 2022, Kry10 Limited.
+# SPDX-License-Identifier: BSD-2-Clause
+
+# Calculate a hash of a directory tree, using file and directory names and
+# contents. Used to uniquely identify graph-refine jobs in CI.
```

Comment:
How long does this one take? An alternative would be something like `fd -x md5sum | md5sum` or something like that (probably doesn't quite work this way, but the idea would be to use fast parallel execution of multiple md5sums and get an overall hash of all the hashes). Might not be worth it, though.

Reply:
The trees I use this on are not that big, so it runs in much less than a second. TBH, a big random number would have worked just as well, since the main purpose is to avoid collisions on job submission.